packages/land/test/common/Royalty.behavior.ts

```diff
@@ -0,0 +1,146 @@
+import {expect} from 'chai';
+import {loadFixture} from '@nomicfoundation/hardhat-network-helpers';
+import {Contract, Signer} from 'ethers';
+
+// eslint-disable-next-line mocha/no-exports
+export function shouldCheckForRoyalty(setupLand, Contract: string) {
+  describe(Contract + ':Royalty', function () {
+    let managerAsRoyaltySetter: Contract,
+      manager: Contract,
+      LandContract: Contract,
+      LandAsMinter: Contract,
```

Comment:
I think that we must stop using mocha `before`.
Is harder to track where the variables are declared and what they are when reading a test.
You cannot have different setups in the same describe block, so you end up grouping the tests (it) by the `before` instead of grouping them by what you want to test.